week2/week2_homework.py

```diff
@@ -0,0 +1,109 @@
+# Q1
+
+def fizz_buzz(num):
+    if num % 3 == 0 and num % 5 == 0 :
+        return "FizzBuzz"
+    elif num % 3 == 0 :
+        return "Fizz"
+    elif num % 5 == 0 :
+        return "Buzz"
+    else : 
+        return num
+
+print(fizz_buzz(65))
+
+# Q2
+
+lst= [5, 10, 20]
+
+def get_item(lst, index):
+    try:
+        return lst[index]
+    except IndexError:
+        return 'The index is out of range.'
+
+print(get_item(lst, 2))
+print(get_item(lst, 4))
+
+# Q3
+
+class Jet:
+    def __init__(self, name, country):
+        self.name = name
+        self.country = country
+
+    def jet_info(self):
+        return f'{self.name} {self.country}'
+
+
+jet1 = Jet("Lufthansa", "Germany")
+print(jet1.jet_info())
+
+# Q5
+
+sample_dic = {'name': 'Nayana', 'surname': 'Kamtekar', 'email': 'abc@gmail.com'}
+
+def is_key_exist(key) :
+    if key in sample_dic :
+        return "The key exist in a dictionary"
+    else :
+        return "The key doesn't exist in a dictionary"
+
+print(is_key_exist('email'))
+print(is_key_exist('address'))
+
+
+# Additional Exercises:
+
+# Q6
+
+def showNumbers(limit):
+    for i in range(limit+1):
+        print(f'{i} {"EVEN" if i%2 == 0 else "ODD"}')
```

Comment:
Good job, but a tiny remark. `{"EVEN" if i%2 == 0 else "ODD"}` is bad practice. You always want to keep program logic and printing separated. Here, you mix them up.

Anyway, long story short, this is better:
```Python
even_or_odd = "EVEN" if i%2 == 0 else "ODD"
print(f'{i} {even_or_odd}')
```

now printing and program logic are well separated